src/pages/music.tsx

```diff
@@ -0,0 +1,222 @@
+
+import Layout from "../components/layout"
+import Hero from "../components/hero"
+import Bio from "../components/bio"
+
+const Students: React.FC<{}> = () => {
+    const data = useStaticQuery<GatsbyTypes.MusicQuery>(graphql`
+        query Music {
+            musicians: allSanityPerson(
+                filter: {
+                    roles: { elemMatch: { slug: { current: { eq: "music" } } } }
+                }
+            ) {
+                nodes {
+                    ...StaffProfile
+                }
+            }
+            mainContent: markdownRemark(
+                fileAbsolutePath: { regex: "/music/main.md$/" }
+            ) {
+                html
+                fields {
+                    frontmattermd {
+                        findOutMoreText {
+                            html
+                        }
+                    }
+                }
+                frontmatter {
+                    title
+                    mainImage {
+                        childImageSharp {
+                            fluid(maxWidth: 1920) {
+                                ...GatsbyImageSharpFluid_noBase64
+                            }
+                        }
+                    }
+                    overlayCaption
+                }
+            }
+            extraContent: markdownRemark(
+                fileAbsolutePath: { regex: "/music/extra.md$/" }
+            ) {
+                html
+                frontmatter {
+                    mainImage {
+                        childImageSharp {
+                            fluid(maxWidth: 1200) {
+                                ...GatsbyImageSharpFluid
+                            }
+                        }
+                    }
+                    title
+                }
+            }
+            releases: markdownRemark(
+                fileAbsolutePath: { regex: "/music/releases.md$/" }
+            ) {
+                frontmatter {
+                    releases {
+                        type
+                        title
+                        coverImage {
+                            childImageSharp {
+                                fluid(maxWidth: 1200) {
+                                    ...GatsbyImageSharpFluid
+                                }
+                            }
+                        }
+                        appleMusicLink
+                        spotifyLink
+                    }
+                }
+            }
+            resources: markdownRemark(
+                fileAbsolutePath: { regex: "/music/resources.md$/" }
+            ) {
+                html
+                frontmatter {
+                    title
+                    resources {
+                        title
+                        files {
+                            name
+                            url
+                        }
+                    }
+                }
+            }
+        }
+    `)
```

Comment:
could we extract this query out of the component function? Will allow for mocking for easier testing and previews in future if we decide to go that way (Apollo has nice mocking stuff).

Reply:
Agree that would be a good refactor. Will hold off on that for now though.